### ISSUE TITLE
Incorrect‌ `attention dropout` and fake `flash_conformer`

### ISSUE THREAD

**Issue by lzqlzzq** [tencent-ailab/MuQ#15]
Thanks for the great model! But I found some problems in the code.

## Incorrect `attention_dropout`

In `src/muq/muq/modules/flash_conformer.py`, the dropout remains even the `.eval()` mode is on, would cause inconsistent output between `eval()` runs.
https://github.com/tencent-ailab/MuQ/blob/28847ea50cd31ac4b8b6a7dacc051ad7d1c7606a/src/muq/muq/modules/flash_conformer.py#L696

It is ought to be something like:
```python
    hidden_states = F.scaled_dot_product_attention(query, key, value, attn_mask=attention_mask, dropout_p=self.dropout_p if self.training else 0.0, is_causal=self.is_causal) 
```

## Fake `flash_conformer`

In `src/muq/muq/models/muq_model.py`, becase of the path structure, in a normal working directory, python interpreter would never resolve the `modules.flash_conformer`. So `is_flash=True` would never work.
<img width="2080" height="1240" alt="Image" src="https://github.com/user-attachments/assets/f15906d9-dfa3-4615-9d21-6e4dcefe7446" />

I proposed a fix in: https://github.com/lzqlzzq/flashMuQ
This fix use `AttentionInterface` in `🤗transformers` to dispatch actual attention implementation, as most modern models do.
I benchmarked the implementation with small batch. It has no numerical difference and wall time reduced to 0.86x, moreover 12x vram saving:
```
muq_large_attention_benchmark device=cuda batch=1 sequence_length=750 dtype=float16
muq_large_attention_backend configured_backend=eager actual_backend=eager allocated_before=607302144 allocated_after=608838144 allocated_delta=1536000 peak_delta=68911104 driver_before=None driver_after=None median_forward_ms=13.592
muq_large_attention_backend configured_backend=sdpa actual_backend=flash_attention allocated_before=607732224 allocated_after=609268224 allocated_delta=1536000 peak_delta=18874368 driver_before=None driver_after=None median_forward_ms=11.646
muq_large_attention_comparison sdpa_over_eager_allocated_delta=1.000000 sdpa_over_eager_forward_time=0.856852
```

**Comment by juhayna-zh** [comment]
Thanks for the detailed report and the fix! We've verified both issues you raised:

1. The `attention_dropout` bug in `eval()` mode is indeed a real problem that would cause inconsistent outputs.
2. The `flash_conformer` import path issue prevents the flash attention path from ever being reached in normal usage.

Your proposed solution using `AttentionInterface` from transformers is elegant and the benchmark results look great. We'd be happy to merge this into the official MuQ codebase.

That said, I'm currently swamped with other commitments, so it may take quite a while before I can get to the actual merge and release. I'll keep this on my radar and will update here once I have bandwidth to integrate it properly. Thanks again for the clean fix!
